modules/shortcodes/gist.php

```diff
@@ -45,16 +119,34 @@ function github_gist_shortcode( $atts, $content = '' ) {
 
 	$id = ( ! empty( $content ) ) ? $content : $atts[0];
 
-	// Parse a URL
-	if ( ! is_numeric( $id ) ) {
-		$id = preg_replace( '#https?://gist.github.com/([a-zA-Z0-9]+)#', '$1', $id );
+	// Parse a URL to get an ID we can use.
+	$gist_info = jetpack_gist_get_shortcode_id( $id );
+	if ( empty( $gist_info['id'] ) ) {
+		return '<!-- Invalid Gist ID -->';
```

Comment:
Do we ever show errors to users who can edit the post for shortcodes? I know we do for some widgets.

Reply:
That's a good point. Done in 54f5d763e11ce80f6576aa26991554cf1dfa73fa